test/testdrive/source-linear-operators.td

```diff
@@ -54,3 +66,228 @@ Query:
+Source materialize.public.data (UID):
+| Project (#0, #1, #3)
+
+Query:
+%0 =
+| Get materialize.public.data (UID)
+| Filter !(isnull(#0))
+| Project (#0)
+| ArrangeBy (#0)
+
+%1 =
+| Get materialize.public.data (UID)
+| Filter !(isnull(#1)), (#3 = 4)
+| Project (#1)
+
+%2 =
+| Join %0 %1 (= #0 #1)
+| | implementation = Differential %1 %0.(#0)
+| Project (#0)
+
+> CREATE MATERIALIZED VIEW mv as SELECT s1.a from data s1, data s2 where s1.a = s2.b and s2.d = 4
+
+> SELECT * FROM mv
+1
+1
+1
+1
+
+> DROP VIEW mv;
+
+? EXPLAIN PLAN FOR SELECT s2.a from data s1, data s2 where s1.a = s2.b and s2.d = 4 and s1.d = 4
+Source materialize.public.data (UID):
+| Filter (#3 = 4)
+| Project (#0, #1, #3)
+
+Query:
+%0 =
+| Get materialize.public.data (UID)
+| Filter !(isnull(#0)), (#3 = 4)
+| Project (#0)
+| ArrangeBy (#0)
+
+%1 =
+| Get materialize.public.data (UID)
+| Filter !(isnull(#1)), (#3 = 4)
+| Project (#0, #1)
+
+%2 =
+| Join %0 %1 (= #0 #2)
+| | implementation = Differential %1 %0.(#0)
+| Project (#1)
+
+> CREATE MATERIALIZED VIEW mv as SELECT s2.a from data s1, data s2 where s1.a = s2.b and s2.d = 4 and s1.d = 4
+
+> SELECT * FROM mv
+1
+2
+
+> DROP VIEW mv;
+
+? EXPLAIN PLAN FOR SELECT s2.c from data s1, data s2 where s1.a = s2.a
+Source materialize.public.data (UID):
+| Filter !(isnull(#0))
+| Project (#0, #2)
+
+Query:
+%0 = Let l0 =
+| Get materialize.public.data (UID)
+| Filter !(isnull(#0))
+
+%1 =
+| Get %0 (l0)
+| Project (#0)
+| ArrangeBy (#0)
+
+%2 =
+| Get %0 (l0)
+| Project (#0, #2)
+
+%3 =
+| Join %1 %2 (= #0 #1)
+| | implementation = Differential %2 %1.(#0)
+| Project (#2)
+
+> CREATE MATERIALIZED VIEW mv as SELECT s2.c from data s1, data s2 where s1.a = s2.a
+
+> SELECT * FROM mv
+3
+5
+3
+2
+3
+2
+
+> DROP VIEW mv;
+
+? EXPLAIN PLAN FOR SELECT * FROM (SELECT a, sum(b) FROM data GROUP BY a UNION ALL SELECT a, (a + c)::numeric FROM data) WHERE a = 1
+Source materialize.public.data (UID):
+| Filter (#0 = 1)
+| Project (#0..#2)
+
+Query:
+%0 = Let l0 =
+| Get materialize.public.data (UID)
+| Filter (#0 = 1)
+
+%1 =
+| Get %0 (l0)
+| Project (#1)
+| Reduce group=()
+| | agg sum(#0)
+| Map 1
+| Project (#1, #0)
+
+%2 =
+| Get %0 (l0)
+| Map i64tonumeric((1 + #2))
+| Project (#0, #4)
+
+%3 =
+| Union %1 %2
+
+> CREATE MATERIALIZED VIEW mv as SELECT * FROM (SELECT a, sum(b) FROM data GROUP BY a UNION ALL SELECT a, (a + c)::numeric FROM data) WHERE a = 1
+
+> SELECT * FROM mv
+1 3
+1 3
+1 4
+
+> DROP VIEW mv;
+
+$ kafka-create-topic topic=data2
+
+$ kafka-ingest format=avro topic=data2 schema=${schema}
+{"a": {"long": 3}, "b": {"long": 2}, "c": null, "d": {"long": 4}}
+{"a": {"long": 2}, "b": {"long": 1}, "c": {"long": 5}, "d": null}
+
+> CREATE SOURCE data2
+  FROM KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'testdrive-data2-${testdrive.seed}'
+  FORMAT AVRO USING SCHEMA '${schema}'
+
+? EXPLAIN PLAN FOR SELECT a, c FROM data EXCEPT ALL SELECT a, c FROM data2 where d is null;
```

Comment:
nit: instead of replicating the definition of every view across the test for explaining it, perhaps we could use `explain view` to make the test a bit more maintainable.